### ISSUE TITLE
Integer Variables are incorrectly set as non-integers by default on new characters

### ISSUE THREAD

**Issue by AlexVild** [AscensionGameDev/Intersect-Engine#947]
**Describe the bug**
For any new character, integer values, by default, are set to some non-zero number. Presumably null or undefined?

By default, integers should be set to the int 0.

**Version Encountered**

Beta 7

**To Reproduce**
1. Create some character, and enter the game world in the client. This will be character A
2. Sign out
3. Create some new integer player variable
4. Create some event that sets the value of that variable to some other variable - i.e, make a counter variable, and set it equal to the variable that you created in step (1)
5. Log the output of the counter variable in the event using chatbox text or some equivalent
6. Create a new character - this will be Character B
7. Trigger the event with Character B, and note the output - it should be some large negative number
8. Now, sign out, and sign in with Character A (who existed before the creation of the player variable). Trigger the event - it will work more as expected, with the variable value defaulting to 0.

**Expected behavior**
I would expect that, by default, any integer player variable is set to 0 on creation of a new character, OR creation of a new variable.

**Screenshots**
![image](https://user-images.githubusercontent.com/8580915/136711884-bfd3ddb2-e993-4e5c-b7fb-44db8e765626.png)

Above is the output from the following event, based on a fresh character

![image](https://user-images.githubusercontent.com/8580915/136711895-c2f39f15-b619-4d81-954c-240ea9650a25.png)

**Operating System:**
Windows

**Comment by Cheshire92** [comment]
I'm sorry, but I can't reproduce this at all.
Can you try making a blank gamedb that has nothing but this problem in it?

I've set up a global variable with a random value that then gets transferred into a regular variable upon interact and it always returns the correct value for me, regardless of what character I do this on, new old or even on a new account.

![Image](https://www.ascensiongamedev.com/resources/filehost/0392e53d91dae2639014f4cea566faab.png)
![Image](https://www.ascensiongamedev.com/resources/filehost/4ffebaa10ff6cc6087696a7beb4a5ec7.png)

**Comment by WeylonSantana** [comment]
@AlexVild I couldn't reproduce it either. I made this:
I downloaded a clean version, started the server from 0
I created a "counter" variable and set it to 30 at login
So I logged in, and I showed the value in the chat, it was 30
I closed the client
I created another variable, and in the event that I set it to 30, I had it set to the value of the new variable created.
When I entered the game it showed 0

I don't know if I did it correctly as you said, maybe I should send a video to show it better?
For now it will be as it is not possible to reproduce
